source/custom/nixnet_service.custom.cpp

```diff
@@ -1187,15 +1187,39 @@ void convert_to_grpc(std::vector<u8>& input, google::protobuf::RepeatedPtrField<
   }
 }
 
-void convert_to_grpc(const void* input, nixnet_grpc::FrameBuffer* output, u32 frame_type)
+void convert_to_grpc(const void* input, nixnet_grpc::FrameBufferResponse* output, u32 protocol, std::map<std::int32_t, std::int32_t> enetflags_output_map)
 {
-  if (frame_type == nixnet_grpc::FrameType::FRAME_TYPE_ENET) {
-    nixnet_grpc::EnetFrame* enet_frame = new nixnet_grpc::EnetFrame();
+  if (protocol == nixnet_grpc::Protocol::PROTOCOL_ENET) {
+    nixnet_grpc::EnetFrameResponse* enet_frame = new nixnet_grpc::EnetFrameResponse();
     nxFrameEnet_t* nxEnetFrame = (nxFrameEnet_t*)input;
-    enet_frame->set_type(nxEnetFrame->Type);
+    enet_frame->set_type(static_cast<nixnet_grpc::EnetFrameType>(nxEnetFrame->Type));
+    enet_frame->set_type_raw(nxEnetFrame->Type);
     enet_frame->set_device_timestamp(nxEnetFrame->DeviceTimestamp);
     enet_frame->set_network_timestamp(nxEnetFrame->NetworkTimestamp);
-    enet_frame->set_flags(nxEnetFrame->Flags);
+    if(nxEnetFrame->Flags & 0x80000000) {
```

Comment:
Note: this is really impossible to review. This is a perfect example of why both codegen and tests are important. Please add it to your retrospective.

Normally, if you were writing this without codegen, you would need a way to implement this with a loop. I believe in this case it would be iterating over the mapping (for unmapped it would be iterating over a list of flags) and checking for each value and then adding it. The "unrolled" implementation makes sense when it comes from a loop in the codegenerator.

Reply:
I agree some unit tests here would be really helpful

Reply:
Yes, I will be adding unit tests in subsequent PRs